packages/stats/gbstats/gbstats.py

```diff
@@ -562,6 +598,46 @@ def process_single_metric(
     )
 
 
+def get_bandit_weights(
+    rows: pd.DataFrame,
+    var_id_map: VarIdMap,
+    metric: MetricSettingsForStatsEngine,
+    analysis: AnalysisSettingsForStatsEngine,
+) -> pd.DataFrame:
+    reduced = preprocess_analysis(rows, var_id_map, metric, analysis)
+    num_variations = reduced.at[0, "variations"]
+    bandit_prior = GaussianPrior(mean=0, variance=float(1e4), proper=True)
+    bandit_config = BanditConfig(prior_distribution=bandit_prior)
+
+    def get_bandit_weights_single(s: pd.Series) -> pd.Series:
+        s_final = pd.Series({"dimension": s["dimension"]})
+        # initialize weights as constant; then just return s if any update checks below fail
```

Comment:
Why return them as constant and equally split? That could be different from what they came in as. I would consider a data structure that allows us to return nothing if some error field is returned. It's better not to return broken data if we don't need to. 

We unfortunately return broken data elsewhere when a test fails (e.g. see the `_default_output` methods), but this is not a great solution we should avoid if we can.

Reply:
changed to return `None` if variation counts are too small or any update checks fail